### PR TITLE
tests: in autopkgtests, use a tempdir in home, not in the tmpfs

### PR DESCRIPTION
**Pull request by elopio**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

File: debian/tests/integrationtests [hunk 1]
```diff
@@ -14,6 +14,7 @@ if [ -z "$SNAPCRAFT_SLOW_TESTS" ]; then
     SNAPCRAFT_SLOW_TESTS=1
 fi
 
+mkdir -p /home/ubuntu/autopkgtest_tmp
 for suite in $suites; do
-    su ubuntu -c "SNAPCRAFT_FROM_INSTALLED=1 TEST_STORE=fake ADT_TEST=1 PATH=/snap/bin:$PATH python3 -m unittest discover -b -v -s ${suite}"
+    su ubuntu -c "SNAPCRAFT_FROM_INSTALLED=1 TEST_STORE=fake ADT_TEST=1 PATH=/snap/bin:$PATH TMPDIR=/home/ubuntu/autopkgtest_tmp python3 -m unittest discover -b -v -s ${suite}"
```
Comment (kyrofa) [inline debian/tests/integrationtests]: Can we not use $HOME here? I guess we're `su`ing as `ubuntu` anyway, but this hard-coded path still bothers me a little.

Comment (elopio) [reply debian/tests/integrationtests]: @kyrofa how do I make sure that $HOME is not expanded to /root?

I tried this, which made me hard-code it:

```
# su ubuntu -c "TMPDIR=$HOME/autopkgtest_tmp python3 -c \"import os; print(os.environ.get('TMPDIR'))\""
/root/autopkgtest_tmp
```

Comment (elopio) [reply debian/tests/integrationtests]: Ohh, this works:
```
# su ubuntu -c "TMPDIR=\$HOME/autopkgtest_tmp python3 -c \"import os; print(os.environ.get('TMPDIR'))\""
/home/ubuntu/autopkgtest_tmp
```
I'll update it.

File: debian/tests/integrationtests [hunk 2]
```diff
@@ -14,6 +14,7 @@ if [ -z "$SNAPCRAFT_SLOW_TESTS" ]; then
     SNAPCRAFT_SLOW_TESTS=1
 fi
 
+mkdir -p /home/ubuntu/autopkgtest_tmp
```
Comment (kyrofa) [inline debian/tests/integrationtests]: You use `$HOME` below, can you do so here?

Comment (kyrofa) [reply debian/tests/integrationtests]: Also, who owns this dir now? root?

Comment (elopio) [reply debian/tests/integrationtests]: Good point. I've added 777 to the dir.

**Comment by sergiusens** [comment]
Please do not update this PR until the adt tests finish running!!!!!

**Pull request merged**